core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -53,7 +53,7 @@ class ReplicaFetcherThread(name: String,
                                 failedPartitions,
                                 fetchBackOffMs = brokerConfig.replicaFetchBackoffMs,
                                 isInterruptible = false,
-                                replicaMgr.brokerTopicStats) {
+                                if (replicaMgr == null) new BrokerTopicStats else replicaMgr.brokerTopicStats) {
```

Comment:
This is only the case in the benchmark code: https://github.com/apache/kafka/blob/trunk/jmh-benchmarks/src/main/java/org/apache/kafka/jmh/fetcher/ReplicaFetcherThreadBenchmark.java#L239

Reply:
Maybe we can fix the benchmark? I'm not sure we should be catering the code to test cases.

Reply:
To me it is more reasonable to change the constructor actually: the `brokerTopicStats` is added this week which breaks the benchmark silently, while I looked at this case I felt that theoretically we should allow the replica fetcher constructor to take either a mock replica manager or a null in order for unit tests and benchmarks that do not require the embedded module but are only testing the logic of the replica fetcher itself. I looked into how much it would take to create a mock replica manager and felt it is a bit too larger of a scope for this PR itself so I make this shortcut fix. If you feel strong about this I think we should be looking into creating a mock replica manager really, instead of letting tests that only want to test the replica fetcher module to initialize a full replica manager.

Reply:
I think we should do this https://github.com/lbradstreet/kafka/commit/8938a6b710634b0c737df32b7070ec8b85f0ed3c. I originally passed null in because I had overridden all of the methods that used it. We need to avoid mocks in anything we want to benchmark, but since it's in the setup this seems fine.

Reply:
Perhaps we can factor out of ReplicaFetcherThread the critical performance sections. Obviously we don't actually support allowing `ReplicaManager` to be null here.